sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java

```diff
@@ -121,42 +123,60 @@ public AWSCredentialsProvider deserializeWithType(
+              new BasicAWSCredentials(
+                  getNotNull(asMap, AWS_ACCESS_KEY_ID, typeName),
+                  getNotNull(asMap, AWS_SECRET_KEY, typeName)));
         }
-      } else if (typeName.equals(PropertiesFileCredentialsProvider.class.getSimpleName())) {
-        return new PropertiesFileCredentialsProvider(asMap.get(CREDENTIALS_FILE_PATH));
-      } else if (typeName.equals(
-          ClasspathPropertiesFileCredentialsProvider.class.getSimpleName())) {
-        return new ClasspathPropertiesFileCredentialsProvider(asMap.get(CREDENTIALS_FILE_PATH));
-      } else if (typeName.equals(DefaultAWSCredentialsProviderChain.class.getSimpleName())) {
+      } else if (hasName(PropertiesFileCredentialsProvider.class, typeName)) {
+        return new PropertiesFileCredentialsProvider(
+            getNotNull(asMap, CREDENTIALS_FILE_PATH, typeName));
+      } else if (hasName(ClasspathPropertiesFileCredentialsProvider.class, typeName)) {
+        return new ClasspathPropertiesFileCredentialsProvider(
+            getNotNull(asMap, CREDENTIALS_FILE_PATH, typeName));
+      } else if (hasName(DefaultAWSCredentialsProviderChain.class, typeName)) {
         return new DefaultAWSCredentialsProviderChain();
-      } else if (typeName.equals(EnvironmentVariableCredentialsProvider.class.getSimpleName())) {
+      } else if (hasName(EnvironmentVariableCredentialsProvider.class, typeName)) {
         return new EnvironmentVariableCredentialsProvider();
-      } else if (typeName.equals(SystemPropertiesCredentialsProvider.class.getSimpleName())) {
+      } else if (hasName(SystemPropertiesCredentialsProvider.class, typeName)) {
         return new SystemPropertiesCredentialsProvider();
-      } else if (typeName.equals(ProfileCredentialsProvider.class.getSimpleName())) {
+      } else if (hasName(ProfileCredentialsProvider.class, typeName)) {
         return new ProfileCredentialsProvider();
-      } else if (typeName.equals(EC2ContainerCredentialsProviderWrapper.class.getSimpleName())) {
+      } else if (hasName(EC2ContainerCredentialsProviderWrapper.class, typeName)) {
         return new EC2ContainerCredentialsProviderWrapper();
-      } else if (typeName.equals(STSAssumeRoleSessionCredentialsProvider.class.getSimpleName())) {
+      } else if (hasName(STSAssumeRoleSessionCredentialsProvider.class, typeName)) {
         return new STSAssumeRoleSessionCredentialsProvider.Builder(
-                asMap.get(ROLE_ARN), asMap.get(ROLE_SESSION_NAME))
+                getNotNull(asMap, ROLE_ARN, typeName),
+                getNotNull(asMap, ROLE_SESSION_NAME, typeName))
             .build();
       } else {
         throw new IOException(
             String.format("AWS credential provider type '%s' is not supported", typeName));
       }
     }
+
+    private String getNotNull(Map<String, String> map, String key, String typeName)
+        throws IOException {
+      String value = map.get(key);
```

Comment:
```
return Preconditions.checkNotNull(map.get(key), String.format("AWS credentials provider type '%s' is missing '%s'", typeName, key))
```
?

Reply:
That was also my first thought, but then decided to stick to IOExceptions to be more consistent with existing error handling in other cases. The only exception throwing a NPE at the moment is when everything deserializes to null. 
But happy to change to checkNotNull / NPEs if you prefer, it's certainly more concise. Let me know.

Reply:
I'd stick with NPE since it's mostly a problem of user configuration than IO connector itself.

Reply:
Arg, this is annoying... now i remember. `checker` then complains about the nullable return values of `checkNotNull` 💩. That means I have to add `@SuppressWarnings({"nullness"})` as well :/

Reply:
ok, just pushed it with nullness warnings suppressed 👍